rag_experiment_accelerator/config/index_config.py

```diff
@@ -83,5 +84,5 @@ def from_index_name(cls, index_name: str, config: Any) -> "IndexConfig":
             generate_title=bool(int(cls.__get_index_value(values[7]))),
             generate_summary=bool(int(cls.__get_index_value(values[8]))),
             override_content_with_summary=bool(int(cls.__get_index_value(values[9]))),
-            embedding_model=config._find_embedding_model_by_name(values[10].strip()),
+            embedding_model=config._find_embedding_model_by_name(values[11].strip()),
```

Comment:
Why isn't the dimension being returned as part of the IndexConfig?
Does this break "round tripping"?

Reply:
The dimensions are already part of the `embedding_model`, they are read from the config file. I only added them to the index name so that we can have two indices with the same model but different number of dimensions, e.g. 

`surface_p-0_cs-1000_o-200_efc-400_efs-400_sp-0_t-0_s-0_oc-0_dim-3072_text-3-large`

and

`surface_p-0_cs-1000_o-200_efc-400_efs-400_sp-0_t-0_s-0_oc-0_dim-256_text-3-large`

Reply:
Rebased with latest changes